CasadiFunctions/createCasadi_Ligaments.m

```diff
@@ -0,0 +1,178 @@
+
+        for j=idx_lig'
+            % test for valid input force-length
+            if ~exist(model_info.ligament_info.parameters(j).stiffness,'file')
+                error(['Unable to find ' model_info.ligament_info.parameters(j).stiffness,...
+                    '. Add the function to ./ModelComponents, or select an existing function for ',...
+                    'the force-length of ligament ' model_info.ligament_info.parameters(j).ligament_name '.'])
+            end
+    
+            % evaluate force-length for dummy motion
+            f_stiffness = str2func(model_info.ligament_info.parameters(j).stiffness);
+            CSA = model_info.ligament_info.parameters(j).cross_section_area;
+            ls = model_info.ligament_info.parameters(j).slack_length;
+            l_lig = model_info.ligament_info.polyFit.DummySamples.lMT(idx_sort,j);
+            d_lig = model_info.ligament_info.polyFit.DummySamples.dM(idx_sort,j,i);
+            F_lig_j = f_stiffness(CSA,ls,l_lig);
+    
+            % ligament length and force for use in post-processing
+            f_L_lig_j = interpolant(['f_L_lig_' model_info.ligament_info.parameters(j).ligament_name],...
+                'bspline',{q_i},l_lig);
+            f_F_lig_j = interpolant(['f_F_lig_' model_info.ligament_info.parameters(j).ligament_name],...
+                'bspline',{q_i},F_lig_j);
+            
+            L_lig_pp_MX(j) = f_L_lig_j(Qs_MX(i));
+            F_lig_pp_MX(j) = f_F_lig_j(Qs_MX(i));
+
+
+            % calculate moment
+            M_lig_j = F_lig_j.*d_lig;
+            M_i = M_i + M_lig_j;
+        end
+    
+        % create look-up table
+        f_lig_single_coord = interpolant(['f_lig_single_coord_',...
+            model_info.ExtFunIO.coord_names.all{i}], 'bspline',{q_i},M_i);
+        
+        % assign to coordinate
+        M_lig_single_MX(i) = f_lig_single_coord(Qs_MX(i));
+    
+    end
+    
+    
+    %% Ligaments that cross multiple coordinates
+    idx_multi_coord = find(sum(model_info.ligament_info.ligament_spanning_multi_coord,2));
+
+    % Assemble polynomial approximation from coefficients
+    qin_SX = SX.sym('qin',model_info.ExtFunIO.jointi.nq.all,1);
+    l_lig_p_SX = SX(model_info.ligament_info.NLigament,1);
+
+    if sum(model_info.ligament_info.ligament_spanning_multi_coord,'all') > 0
+        ligament_spanning_info_m = model_info.ligament_info.ligament_spanning_multi_coord(:,:);
+        LigamentInfo_m.muscle = model_info.ligament_info.polyFit.LigamentInfo.muscle(:);
+        load nCoeffMat
+        load expoVal_all
+        for i=1:model_info.ligament_info.NLigament      
+            index_dof_crossing = find(ligament_spanning_info_m(i,:)==1);
+            nr_dof_crossing = length(index_dof_crossing);
+            if nr_dof_crossing == 0
+                continue
+            end
+            order = LigamentInfo_m.muscle(i).order;
+            [mat,~] = n_art_mat_3_cas_SX_7(qin_SX(index_dof_crossing,1)',...
```

Comment:
can you add some comments here. I'm not following what is going on here (I assume that you take the derivative of the polynomial expression to each dof ?)